text_localization_environment/TextLocEnv.py

```diff
@@ -205,8 +241,31 @@ def taller(self):
         self.adjust_bbox(np.array([0.5, 0, -0.5, 0]))
 
     def trigger(self):
-        self.done = True
-        #self.create_ior_mark()
+        if self.mode == 'train':
```

Comment:
I see 2 issues with this:

1) This never terminates. You won't be able to train the agent like this because it never sees a second image & just gets stuck on the same image when all instances have been found. For the environment to work properly, we need to introduce some termination condition, e.g. after a maximum number of steps per image, when a loop is detected (or the "all instances found" trigger).

2) I'm not a fan of discarding my original masking implementation (the one that masks 0...N-1 images randomly & trains for one episode per image only). The reason is that we have a working model for this implementation - we don't know if & how fast the model converges when we use the same (your) approach for testing & training.

My suggestion would be to introduce environment subclasses using the template method pattern (for trigger & reset). This way we can maintain & compare different strategies for the environment. Following this approach there would be 3 environments:
1) Base environment that only works for single instances
2) Subclass environment that uses random masking/single episodes (my approach) during training & your approach during testing
3) Another subclass environment that uses your approach for both training & testing (with a proper termination condition so training actually works)